### PR TITLE
Add support for default Class or KClass values

### PR DESCRIPTION
**Pull request by AlexBeggs**
- add support after blocking issue https://github.com/google/ksp/issues/53 was fixed
- remove NPE catch
- add tests for default KClass and Class for getAnnotationsByType
- handle Java Class default value retrieval using reflection for PsiImmediateClassType

### CODE REVIEW

File: compiler-plugin/src/test/kotlin/com/google/devtools/ksp/processor/AnnotatedUtilProcessor.kt [hunk 1]
```diff
@@ -86,9 +86,6 @@ annotation class ParametersTestAnnotation(
     val intValue: Int = 6,
     val longValue: Long = 7L,
     val stringValue: String = "emptystring",
-    // fails on getting the arguments from the KSAnnotation when no value is set for the kClassValue in
-    // the declaration. Throws an NPE with using a default value
-    // https://github.com/google/ksp/issues/53
```
Comment (AlexBeggs) [inline compiler-plugin/src/test/kotlin/com/google/devtools/ksp/processor/AnnotatedUtilProcessor.kt:91]: comments no longer valid

File: compiler-plugin/testData/api/annotatedUtil.kt [hunk 2]
```diff
@@ -56,8 +56,6 @@ annotation class ParametersTestAnnotation(
     val intValue: Int = 6,
     val longValue: Long = 7L,
     val stringValue: String = "emptystring",
-    // fails on getting the arguments from the KSAnnotation when no value is set for the kClassValue in
-    // the declaration. Throws an NPE with using a default value
```
Comment (AlexBeggs) [inline compiler-plugin/testData/api/annotatedUtil.kt:60]: comments no longer valid

File: compiler-plugin/testData/api/annotatedUtil.kt [hunk 3]
```diff
@@ -120,8 +118,8 @@ class OnlyTestAnnotation
 @Test
 class ParametersTestAnnotationWithValuesTest
 
-@ParametersTestAnnotation(kClassValue = ParametersTestAnnotation::class)
-@ParametersTestAnnotation(kClassValue = ParametersTestAnnotation::class)
```
Comment (AlexBeggs) [inline compiler-plugin/testData/api/annotatedUtil.kt:124]: removed the values to test default values

File: compiler-plugin/testData/api/javaAnnotatedUtil.kt [hunk 4]
```diff
@@ -56,8 +56,6 @@ public @interface ParametersTestAnnotation {
     Int intValue() default 6;
     Long longValue() default 7L;
     String stringValue() default "emptystring";
-    // fails on getting the arguments from the KSAnnotation when no value is set for the kClassValue in
-    // the declaration. Throws an NPE with using a default value
```
Comment (AlexBeggs) [inline compiler-plugin/testData/api/javaAnnotatedUtil.kt:60]: comments no longer valid

File: compiler-plugin/testData/api/javaAnnotatedUtil.kt [hunk 5]
```diff
@@ -121,8 +119,8 @@ public class OnlyTestAnnotation {}
 @Test
 public class ParametersTestAnnotationWithValuesTest {}
 
-@ParametersTestAnnotation(kClassValue = ParametersTestAnnotation.class)
-@ParametersTestAnnotation(kClassValue = ParametersTestAnnotation.class)
```
Comment (AlexBeggs) [inline compiler-plugin/testData/api/javaAnnotatedUtil.kt:125]: removed the values to test default values

**Review by ting-yuan — APPROVED** [review]
Let's merge this and investigate the issue separately.

**Pull request merged**